File: C#/QuickSort.cs
using System;

public class QuickSort
{
    public QuickSort()
    {
        int[] numbers = { 9, 7, 5, 11, 12, 2, 14, 3, 10, 6 };
        Console.WriteLine("Unsorted array:");
        PrintArray(numbers);

        QuickSortAlgorithm(numbers, 0, numbers.Length - 1);

        Console.WriteLine("Sorted array:");
        PrintArray(numbers);
    }

    private void QuickSortAlgorithm(int[] array, int low, int high)
    {
        if (low < high)
        {
            int pivotIndex = Partition(array, low, high);
            QuickSortAlgorithm(array, low, pivotIndex - 1);
            QuickSortAlgorithm(array, pivotIndex + 1, high);
        }
    }

    private int Partition(int[] array, int low, int high)
    {
        int pivot = array[high];
        int i = low - 1;

        for (int j = low; j < high; j++)
        {
            if (array[j] <= pivot)
            {
                i++;
                Swap(array, i, j);
            }
        }
        Swap(array, i + 1, high);
        return i + 1;
    }

    private void Swap(int[] array, int i, int j)
    {
        int temp = array[i];
        array[i] = array[j];
        array[j] = temp;
    }

    private void PrintArray(int[] array)
    {
        foreach (int value in array)
        {
            Console.Write(value + " ");
        }
        Console.WriteLine();
    }
}

File: Python/Collection/QuickSort.py
def quick_sort(array, low, high):
    if low < high:
        pivot_index = partition(array, low, high)
        quick_sort(array, low, pivot_index - 1)
        quick_sort(array, pivot_index + 1, high)

def partition(array, low, high):
    pivot = array[high]
    i = low - 1

    for j in range(low, high):
        if array[j] <= pivot:
            i += 1
            swap(array, i, j)
    swap(array, i + 1, high)
    return i + 1

def swap(array, i, j):
    array[i], array[j] = array[j], array[i]

def print_array(array):
    print(array)

# Hauptfunktion
if __name__ == "__main__":
    numbers = [9, 7, 5, 11, 12, 2, 14, 3, 10, 6]
    print("Unsortiertes Array:")
    print_array(numbers)

    quick_sort(numbers, 0, len(numbers) - 1)

    print("Sortiertes Array:")
    print_array(numbers)
